### PR TITLE
Unhashed files

### PR DESCRIPTION
**Pull request by JakeSiFive**
This change adds support for tracking unhashed files in wake so that they can be removed via --clean and listed via --list-outputs.

### CODE REVIEW

File: src/runtime/database.cpp
Comment (mmjconolly) [reply src/runtime/database.cpp:37]: ah it was https://github.com/sifive/wake/blob/master/tools/wake/main.cpp#L385

Comment (JakeSiFive) [reply src/runtime/database.cpp:37]: Yeah it will error out and require wake.db to be deleted. This change isn't actually incompatible though so maybe we should leave it at 5?

Comment (JakeSiFive) [reply src/runtime/database.cpp:37]: Yeah so if a wake that creates the current version uses a current database it will just ignore the unhashed files like it does today. If a wake with this version uses an old database it will just not know about some of the unhashed files but it will start adding any new ones.

Comment (JakeSiFive) [reply src/runtime/database.cpp:37]: Also that line checks a different thing I think, the check for version of the database happens here: https://github.com/sifive/wake/blob/master/src/runtime/database.cpp#L163

**Review by V-FEXrt — APPROVED** [review]
Please wait for review from other folks but from the C++ perspective, LGTM

Also, as an aside, it would be nice to start pulling sql code out of the c/cpp files. I understand why it wasn't done here but its something I'd like to start moving towards

File: src/runtime/database.cpp
Comment (mmjconolly) [inline src/runtime/database.cpp:907]: spelling; compremise

Comment (terpstra) [reply src/runtime/database.cpp:907]: This handles the empty string differently. I'd check that this does not matter.

**Review by terpstra — APPROVED** [review]
You'll still have zombie files (both unhashed and hashed), but that's unavoidable without a database that can track two invocations of the same Job.

**Comment by JakeSiFive** [comment]
ok we've worked out that this functions as expected so I'm going to land this.

**Pull request merged**
